travis.sh

```diff
@@ -45,6 +45,42 @@ ROSWS=wstool
 
 source ${CI_SOURCE_PATH}/$CI_PARENT_DIR/util.sh
 
+# Building in 16.04 requires running this script in a docker container
+# The Dockerfile in this repository defines a Ubuntu 16.04 container
+if [[ "$ROS_DISTRO" == "kinetic" ]] && ! [ "$IN_DOCKER" ]; then
+  travis_time_start build_docker_image
+  docker build -t industrial-ci/xenial .ci_config
```

Comment:
Using dockerhub to automatically update the image whenever there is a downstream change might also save you a lot of build time